src/UCTSearch.cpp

```diff
@@ -519,22 +519,31 @@ size_t UCTSearch::prune_noncontenders(int elapsed_centis, int time_for_move) {
 }
 
 bool UCTSearch::have_alternate_moves(int elapsed_centis, int time_for_move) {
-    if (cfg_timemanage != TimeManagement::ON) {
+    if (cfg_timemanage == TimeManagement::OFF) {
         return true;
     }
     auto pruned = prune_noncontenders(elapsed_centis, time_for_move);
-    if (pruned == m_root->get_children().size() - 1) {
-        // In a timed search we will essentially always exit because
-        // the remaining time is too short to let another move win, so
-        // avoid spamming this message every move. We'll print it if we
-        // save at least half a second.
-        if (time_for_move - elapsed_centis > 50) {
-            myprintf("%.1fs left, stopping early.\n",
-                     (time_for_move - elapsed_centis) / 100.0f);
+    if (pruned > m_root->get_children().size() - 1) {
```

Comment:
typo - should be <

Reply:
Well, <= or < size(), I guess.

Reply:
<= or < size(), I think.

Reply:
Wouldn't that mean that all children need to be pruned to stop? That can't happen because the child with the most visits will always have enough to survive the pruning.